### PR TITLE
front-end/router: updates router to use navigo

### PR DESCRIPTION
**Pull request by forschoolandnothingelse**
Updates the frontend routing to use more conventional routes. While
hash-based routing is still used, routes now look more like standard
URIs.

The new routes are:
/dailyLog/:id
/monthlyLog/:id
/futureLog/:id
/collection/:id
Which look like boojo.app/#/dailyLog/BCykNlF2a4MCiIvKfQJCaC66JvzEZg

The router.js file is refactored to only export the Navigo router
object. This is needed in some cases to change the route using
javascript. The preferred way to change the route will always be using
HTML. The way to do this is by using the <a> element with the
data-navigo attribute.

Example:
`<a href="/dailyLog/BCykNlF2a4MCiIvKfQJCaC66JvzEZg" data-navigo>`

Link to the Navigo documentation:
https://github.com/krasimir/navigo/blob/master/DOCUMENTATION.md

### CODE REVIEW

File: src/front-end/components/navbar.jsx [hunk 1]
```diff
@@ -156,13 +155,11 @@ export class NavBar extends HTMLElement {
 	 * Goes to home page when the home button is pressed
 	 */
 	goHome () {
-		if (document.location.hash !== null && document.location.hash !== "#index" && document.location.hash !== "") {
-			router.setState("", false);
-		}
+		router.navigate("/");
```
Comment (ehsonp) [inline src/front-end/components/navbar.jsx:156]: might reload the page if we are already on it. I think its fine just noting it

File: src/front-end/index.js [hunk 2]
```diff
@@ -49,11 +45,3 @@ document.getElementById("sidebar").appendChild(navbar);
 document.getElementById("targetMenu").onclick = () => {
 	navbar.toggleTracker();
 };
-
-
-// Document.getElementById("")
-router.setState(document.location.hash, false);
-
```
Comment (ehsonp) [inline src/front-end/index.js:56]: How are we able to not set state in index when its ran

**Review by ehsonp — APPROVED** [review]
Looks good. I tested it and it maintains functionality with a great routing improvement

File: src/front-end/index.js (hunk 2 above)
Comment (ehsonp) [reply src/front-end/index.js:56]: I get it, nevermind

**Review by e5chang — APPROVED** [review]

**Pull request merged**